ROADMAP.md

```diff
@@ -0,0 +1,89 @@
+# Chisel Roadmap
+
+This roadmap captures the current plans of the Chisel maintainers and can be
+changed through the usual PR process.
+Not every small feature add or planned bugfix is logged here, but the high-level
+changes and planned direction that are relevant for those using and contributing
+to the project.
+Two especially notable directions on the roadmap below are:
+
+* We are migrating Chisel to rely on, exclusively, the
+  [MLIR-based FIRRTL Compiler](https://github.com/llvm/circt)
+* Accelerating improvements and changes to Chisel and the FIRRTL specification
+
+
+## 3.5.x (and Earlier)
+
+- Continue to maintain binary compatibility on these releases
+- Continue to backport bug fixes, new features if possible, documentation
+  updates, etc.
+- Deprecations are backported to ease in users migrating to next major release
+  (3.6).
+
+## 3.6 (Target 2022 November)
+
+This release is intended to provide a smooth path for the Chisel codebase to no
+longer depend on the 
+[Scala FIRRTL Compiler (SFC)](https://github.com/chipsalliance/firrtl)
+and use the [MLIR Firrtl Compiler (MFC)](https://github.com/llvm/circt) as the
+main backend. While the SFC may continue to work to consume the `.fir` produced
+by Chisel, it will not be a requirement for adding new features to the FIRRTL
+spec and the Chisel APIs. Thus, the 3.6 release will:
+
+- Deprecate `Chisel._`
+- Turn the deprecations from previous releases into Errors
+- Support integration with [MFC](https://github.com/llvm/circt) by bringing
+  `chisel-circt` in-repo
+- Stop using [SFC](https://github.com/chipsalliance/firrtl) except to test
+  deprecated features
+- Deprecate all Chisel features that are not implemented in MFC. A nonexhaustive
+  list includes:
+  - Interval Types
+  - Fixed Point Types
+  - Injecting Aspects
+  - EnumAnnotations
```

Comment:
Would you consider keeping these around until you add enums to the firrtl IR?
Essentially in the same PR where you switch chisel to emitting the native firrtl enums would be the appropriate place for removing the old annotation based system. This way we do not loose any information in the meantime.

Reply:
I have a FSM coverage pass that relies on the enum annotations to discover state machine registers and possible state transitions.

Reply:
I think it is reasonable to allow Chisel to emit whatever annotations it wants. However, an `EnumAnnotation` is not updated or understood by a FIRRTL compiler. Put differently, this is a move towards FIRRTL compilers having an allow-list of known annotations and `EnumAnnotation` would not be allowed or, more weakly, any `EnumAnnotation` is not output from a FIRRTL compiler. If a user wants this style of behavior, they should use `chisel3.experimental.Trace`.

Reply:
So firrtl compilers won't be required by the spec to support user defined annotations, but if a specific firrtl compiler choose to do so, it would still be able to get the annotation out of Chisel, correct? That would sound reasonable to me.